### ISSUE TITLE
peerDependencies for serverless use ~ range specifier rather than ^ specifier

### ISSUE THREAD

**Issue by wrumsby** [newrelic/serverless-newrelic-lambda-layers#274]
Hello,

This is not a "bug" per se, but perhaps an "issue"

Currently the peer dependencies for this package include:

```
    "@types/serverless": "~3.12.7",
    "serverless": "~3.19.0"
```

This suggests that only  "valid" peer dependency versions of `serverless` are 3.19.x.

Since npm 7, npm has reverted to uninstalling missing peer dependencies which means that other versions of of serverless 3.x.x will not be considered matches and may mean that npm tries to install multiple versions of serverless or perhaps fails.

Given the other peer dependencies use a `^` range specifier I suspect there might be a valid reason for the use of the more conservative `~` range specifier for `serverless`, but I'd like to express my _concern_ that this might lead to an undesirable and confusing developer experience.

This problem is likely to manifest with node 16 which is bundled with npm 8.

**Comment by Plitzer** [comment]
+1
At the moment we have to tie our serverless version to whatever *minor* version this plugin says it supports, which is not ideal. Trying to install the latest serverless version (3.23.0 at time of writing) causes the install to error.

**Comment by mrickard** [comment]
Fixed with v3.3.6. Thank you, @ran-j !
